src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Llm/EventHelper.cs

```diff
@@ -0,0 +1,205 @@
+// Copyright 2020 New Relic, Inc. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.Collections.Generic;
+using NewRelic.Agent.Api;
+
+namespace NewRelic.Agent.Extensions.Llm
+{
+    public static class EventHelper
+    {
+        public static string CreateChatCompletionEvent(IAgent agent,
+            ISegment segment,
+            string requestId,
+            float temperature,
+            int maxTokens,
+            string requestModel,
+            string responseModel,
+            int numMessages,
+            string finishReason,
+            string vendor,
+            bool isError,
+            IDictionary<string, string> headers,
+            LlmErrorData errorData)
+        {
+            var completionId = Guid.NewGuid().ToString();
+
+            var attributes = new Dictionary<string, object>
+            {
+                { "id", completionId },
+                { "request_id", requestId },
+                { "span_id", segment.SpanId },
+                { "trace_id", agent.GetLinkingMetadata()["trace.id"] },
+                { "request.temperature", temperature },
+                { "request.max_tokens", maxTokens },
+                { "request.model", requestModel },
+                { "response.model", responseModel },
+                //{ "response.organization", "not available" },
+                { "response.number_of_messages", numMessages },
+                { "response.choices.finish_reason", finishReason },
+                { "vendor", vendor },
+                { "ingest_source", "DotNet" },
+                { "duration", (float)segment.DurationOrZero.TotalMilliseconds },
+                //{ "llm.<user_defined_metadata>", "Pulled from Transaction metadata in RecordLlmEvent" },
+                //{ "response.headers.<vendor_specific_headers>", "See LLM headers below" },
```

Comment:
Should we remove the commented out code?

Reply:
I had the same thought, but I also found that reading through the code to understand the comments helped me understand the overall structure of things, so I don't mind leaving them in.